Mathlib/Topology/Algebra/Valued/LinearTopology.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2025 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Topology.Algebra.LinearTopology
+import Mathlib.Topology.Algebra.Valued.ValuedField
+
+/-!
+# Valuation rings of valued fields have a linear topology
+
+## Main Results
+* `IsLinearTopology.of_valued`: for a valued field `K`,
+  the valuation ring `𝒪[K]` has a linear topology
+
+-/
+open Valued Filter Topology
+
+variable {K Γ₀ : Type*} [Field K] [LinearOrderedCommGroupWithZero Γ₀] [Valued K Γ₀]
+
+instance IsLinearTopology.of_valued :
+    IsLinearTopology 𝒪[K] 𝒪[K] := by
+  by_cases hd : DiscreteTopology K
```

Comment:
This proof needs some comments to explain the logic; there's a rule of thumb that you shouldn't usually have more than 10 lines of tactic code in a row without a comment.